backend/main.go

```diff
@@ -183,20 +184,29 @@ func enhancedHealthCheck(ctx context.Context, db *gorm.DB, rClient *redis.Client
 	}
 }
 
+var PUBLIC_GRAPH_CORS_OPTIONS = cors.Options{
+	AllowedOrigins:   []string{"*"},
+	AllowCredentials: false,
+	AllowedHeaders:   []string{"*"},
+}
+
+var PRIVATE_GRAPH_CORS_OPTIONS = cors.Options{
+	AllowOriginRequestFunc: validateOrigin,
+	AllowCredentials:       true,
+	AllowedHeaders:         []string{"*"},
+}
+
 func validateOrigin(_ *http.Request, origin string) bool {
-	if runtimeParsed == util.PrivateGraph {
-		// From the highlight frontend, only the url is whitelisted.
-		isRenderPreviewEnv := strings.HasPrefix(origin, "https://frontend-pr-") && strings.HasSuffix(origin, ".onrender.com")
-		// Is this an AWS Amplify environment?
-		isAWSEnv := strings.HasPrefix(origin, "https://pr-") && strings.HasSuffix(origin, ".d25bj3loqvp3nx.amplifyapp.com")
-		isReflamePreview := origin == "https://preview.highlight.io"
-
-		if origin == frontendURL || origin == "https://app.highlight.run" || origin == "https://app.highlight.io" || origin == landingStagingURL || isRenderPreviewEnv || isAWSEnv || isReflamePreview {
-			return true
-		}
-	} else if runtimeParsed == util.PublicGraph || runtimeParsed == util.All {
```

Comment:
> we should probably have separate public/private configs for hobby deploys too rather than using a single one for the All runtime?

yeah, we can detect the hobby deploy via the `util.IsInDocker()` check, split out the configs and then use the `REACT_APP_PRIVATE_GRAPH_URI` and `REACT_APP_PUBLIC_GRAPH_URI` values for the respective origin checks. with the normal hobby deploy tho, the private and public graphs run together, but assuming the runtime flag is used we should use the corresponding origin